modules/light-clients/08-wasm/types/client_state.go

```diff
@@ -116,19 +112,11 @@ func (cs ClientState) Initialize(ctx sdk.Context, _ codec.BinaryCodec, clientSto
 		ConsensusState: consensusState,
 	}
 
-	encodedData, err := json.Marshal(payload)
-	if err != nil {
-		return errorsmod.Wrapf(err, "failed to marshal payload for wasm contract instantiation")
-	}
-
 	// The global store key can be used here to implement #4085
 	// wasmStore := ctx.KVStore(WasmStoreKey)
 
-	_, err = initContract(ctx, clientStore, cs.CodeHash, encodedData)
-	if err != nil {
-		return errorsmod.Wrapf(err, "failed to initialize contract")
-	}
-	return nil
+	err := wasmInit(ctx, clientStore, &cs, payload)
+	return err
```

Comment:
this could be `return wasmInit(..)` right? 

my preference would be either for checking the err above and returning nil on 119 or else a direct return of `wasmInit()`

Reply:
went for the direct return for now, nice catch :pray: